### PR TITLE
SCP-1543: Marlowe Playground Frontend: Implement redesign of save as

### PR DESCRIPTION
**Pull request by hrajchert**
This PR implements the redesign for the "Save as" modal
Screen Shot 2020-12-11 at 10 53 00

Screen Shot 2020-12-11 at 10 53 07

Screen Shot 2020-12-15 at 11 46 46

Screen Shot 2020-12-15 at 11 46 28

Pre-submit checklist:
- Branch
    - [X] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested
- If you changed any Purescript files:
   - [X] `$(nix-shell shell.nix --run fix-purty)` to fix any formatting issues

Pre-merge checklist:
- [x] Someone approved it
- [x] Commits have useful messages
- [x] Review clarifications made it into the code
- [x] History is moderately tidy; or going to squash-merge

### CODE REVIEW

**Review by merivale — APPROVED** [review]
Looks good to me. Just a couple of notes (take them or leave them):

1. You've got a new `.saved-as-modal` class, which I think is a typo? Should presumably be `.save-as-modal`. Also, just wondering if this class name is a bit specific...? But that depends on what other modals you have planned. 🤷‍♂️ Up to you.
2. I also suggest giving the two buttons a fixed width (~8em ought to do it). I think it's nicer if they're the same width, but more importantly you don't want the "save" button to shrink when it's saving and the text changes to an icon (as it does at the moment).

**Comment by hrajchert** [comment]
Thanks @merivale ! I applied both suggestions.

The class `save-as-modal` is intended to be super specific and not to be reused

File: marlowe-playground-client/static/css/css_var_colors.scss [hunk 1]
```diff
@@ -2,7 +2,7 @@
   --white: white;
   --black: black;
   --primary-color: #832dc4;
-  --secondary-color: #1746a0;
+  --secondary-color: #1746a0; // TODO: replace with --blue
```
Comment (shmish111) [inline marlowe-playground-client/static/css/css_var_colors.scss:5]: why not do this work in this PR?

**Review by shmish111 — APPROVED** [review]
LGTM, just one question but maybe it can be in another PR.

File: marlowe-playground-client/static/css/css_var_colors.scss (hunk 1 above)
Comment (hrajchert) [reply marlowe-playground-client/static/css/css_var_colors.scss:5]: I wanted to confirm the approach of having --blue-xxx in this PR before going ahead and modify all instances of this colors. Will do in a separate PR

**Pull request merged**